browser/brave_stats_updater.cc

```diff
@@ -100,19 +98,16 @@ void BraveStatsUpdater::Stop() {
 
 void BraveStatsUpdater::OnSimpleLoaderComplete(
     std::unique_ptr<brave::BraveStatsUpdaterParams> stats_updater_params,
-    std::unique_ptr<std::string> response_body) {
+    scoped_refptr<net::HttpResponseHeaders> headers) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
   int response_code = -1;
-  if (simple_url_loader_->ResponseInfo() &&
-      simple_url_loader_->ResponseInfo()->headers)
-    response_code =
-        simple_url_loader_->ResponseInfo()->headers->response_code();
+  if (headers)
+    response_code = headers->response_code();
   if (simple_url_loader_->NetError() != net::OK || response_code < 200 ||
       response_code > 299) {
     LOG(ERROR) << "Failed to send usage stats to update server"
                << ", error: " << simple_url_loader_->NetError()
                << ", response code: " << response_code
-               << ", payload: " << *response_body
```

Comment:
@pilgrim-brave I guess this line was the root cause of the crash?

Reply:
No, the crash was earlier. If SimpleURLLoader received a network error with a large response body, it attempted to copy the entire body into a fixed length buffer to call the OnSimpleLoaderComplete callback function. With a large enough body, this crashed. Fix uses DownloadHeadersOnly to ignore the response body altogether.

Reply:
OK I see, thanks for catching that!